src/pke/lib/scheme/ckksrns/ckksrns-schemeswitching.cpp

```diff
@@ -637,17 +741,14 @@ std::vector<std::vector<NativeInteger>> ExtractLWEpacked(const Ciphertext<DCRTPo
     auto originalB{(ct->GetElements()[0]).GetElementAtIndex(0)};
     originalA.SetFormat(Format::COEFFICIENT);
     originalB.SetFormat(Format::COEFFICIENT);
-    auto N = originalB.GetLength();
 
-    std::vector<std::vector<NativeInteger>> extracted(2);
-    extracted[0].reserve(N);
-    extracted[1].reserve(N);
+    size_t N = originalB.GetLength();
 
-    auto& originalAVals = originalA.GetValues();
-    auto& originalBVals = originalB.GetValues();
-
-    extracted[1].insert(extracted[1].end(), &originalAVals[0], &originalAVals[N]);
-    extracted[0].insert(extracted[0].end(), &originalBVals[0], &originalBVals[N]);
+    // create 2 "begin" iterators to work with element values
+    const auto itA = std::vector<NativeInteger>::const_iterator(&originalA.GetValues()[0]);
+    const auto itB = std::vector<NativeInteger>::const_iterator(&originalB.GetValues()[0]);
+    std::vector<std::vector<NativeInteger>> extracted{std::vector<NativeInteger>(itB, itB + N),
+                                                      std::vector<NativeInteger>(itA, itA + N)};
 
```

Comment:
This is correct as written, but I would suggest changing L740-741 to
```
    auto originalA{(ct->GetElements()[0]).GetElementAtIndex(0)};
    auto originalB{(ct->GetElements()[1]).GetElementAtIndex(0)};
```
so that this becomes
```
    std::vector<std::vector<NativeInteger>> extracted{std::vector<NativeInteger>(itA, itA + N),
                                                      std::vector<NativeInteger>(itB, itB + N)};
```